packages/client/src/runtime/core/engines/data-proxy/DataProxyEngine.ts

```diff
@@ -162,32 +161,39 @@ export class DataProxyEngine extends Engine<DataProxyTxInfoPayload> {
     this.clientVersion = config.clientVersion ?? 'unknown'
     this.logEmitter = config.logEmitter
     this.tracingHelper = this.config.tracingHelper
+  }
+
+  apiKey(): string {
+    return this.headerBuilder.apiKey
+  }
+
+  version() {
+    // QE is remote, we don't need to know the exact commit SHA
+    return 'unknown'
+  }
 
+  /**
+   * This is not a real "start" method, but rather a deferred initialization. We
+   * will only parse the URL on the first request to match the behavior of other
+   * engines, they will only throw errors on their very first request. This is
+   * needed in case the URL is misconfigured.
+   */
+  async start() {
```

Comment:
The URL validation and extraction was moved into `start`, then `url` needs this to be able to calculate the Accelerate URL.